### PR TITLE
Added a solution to the challenge 'groupBy' in CSX/Callbacks

### PR DESCRIPTION
**Pull request by a-hend**
Please review if you have time.

NOTE:  I just noticed that I added more elements to the array 'decimals' than the original challenge included, so it won't log what line 20 says it should (but that might have been obvious to you :-)

### CODE REVIEW

**Review by C-STYR — CHANGES_REQUESTED** [review]
Nice use of `forEach`.  Did you know it has an optional index param? (you don't need it here...)
In general, computations are expensive and we only want to run them once. By a computation, I mean anytime you're running a function. So, we can save the value of `callback(el)` to a variable, and then replace all subsequent `callback(el)` uses with that variable.
There is no reason to have a `cache` array at all; you already have an array initialized in the form of `[el]`. Just push in there.

**Comment by C-STYR** [comment]
So, I've requested changes. The way to deal with that is to make changes locally, push up another commit to this branch, and then click the down arrow next to the line that says "1 change requested" and then click "re-request review". Good luck!

**Comment by a-hend** [comment]
Ok, I've done my best.

**Comment by a-hend** [comment]
Using `[el]` as a variable that way totally worked.  It took me a few minutes to figure out how to do it and I will need some more practice until I get comfortable with it, but it really cleaned things up.  Thanks!

File: Users/abehenderson/CSX/Callbacks/groupBy.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+// Attempt #2 
+  // Per Cole's Change Request:
+    // Initialize a variable equal to invoking the callback on 'el'
+    // Eliminate 'cache' and use '[el]' to temporarily store values
+
+function groupBy(arrays, callback) {
+  const obj = {};
+  arrays.forEach(el => {
+    let evRes = callback(el); // Cole: what is a better name that 'evRes'?
```
Comment (C-STYR) [inline Users/abehenderson/CSX/Callbacks/groupBy.js]: how about `key`? Fundamentally that's what you're creating: the keys of this object, which are the returned values of `callback(el)`.

File: Users/abehenderson/CSX/Callbacks/groupBy.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+// Attempt #2 
+  // Per Cole's Change Request:
+    // Initialize a variable equal to invoking the callback on 'el'
+    // Eliminate 'cache' and use '[el]' to temporarily store values
```
Comment (C-STYR) [inline Users/abehenderson/CSX/Callbacks/groupBy.js]: ...not temporarily

File: Users/abehenderson/CSX/Callbacks/groupBy.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+function groupBy(arrays, callback) {
+  const obj = {};
+  arrays.forEach(el => {
+    let evRes = callback(el); // Cole: what is a better name that 'evRes'?
+    if (!obj[evRes]) {
+      obj[evRes] = [el];
+    } else {
+      [el] = obj[evRes]; // I didn't know I could use '[el]' as a variable like this
```
Comment (C-STYR) [inline Users/abehenderson/CSX/Callbacks/groupBy.js]: you've gotten into some advanced shit here by accident!
there is a much simpler way. 
on line 11, you've assigned a value to the `evRes` key on the `obj` object.  the value is the element `el` inside an array. 
the directions want you to add further elements to THAT ARRAY if they meet the correct criterium. you don't need line 13 at all; you don't know it but all you're doing there is what's called "destructuring" the value of `obj[evRes]` and assigning the `obj[evRes][0]` to variable `el`. the reason your code works with line 13 included is that it has no effect on line 14.

**Review by C-STYR — CHANGES_REQUESTED** [review]

File: Users/abehenderson/CSX/Callbacks/groupBy.js (hunk 1 above)
Comment (a-hend) [reply Users/abehenderson/CSX/Callbacks/groupBy.js]: Ok, of course!  That seems rather obvious now that you mention it.

File: Users/abehenderson/CSX/Callbacks/groupBy.js (hunk 3 above)
Comment (a-hend) [reply Users/abehenderson/CSX/Callbacks/groupBy.js]: Yes, I see that now.  

I was stuck in the mindset that I needed with the previous way I solved the problem.  I was using `cache` as temporary storage that I would fill and dump with each iteration.  That idea got stuck in my head and carried over, even when you said I didn't need `cache`

Comment (a-hend) [reply Users/abehenderson/CSX/Callbacks/groupBy.js]: But here I can see that once `obj[evRes] = [el];` is declared, I can just continue to push to the array `[el]` without the need for temporary storage....am I understanding that correctly?

**Comment by a-hend** [comment]
This all suddenly seems so much simpler than the way I had been thinking about it.  I didn't know I could push directly to an array established as the value of an object key `[el]`, so I was basically creating a work around.

File: Users/abehenderson/CSX/Callbacks/groupBy.js (hunk 2 above)
Comment (a-hend) [reply Users/abehenderson/CSX/Callbacks/groupBy.js]: roger, I get it now

**Review by C-STYR — APPROVED** [review]
Nice!

**Pull request merged**